build.gradle.kts

```diff
@@ -1,382 +1,40 @@
-@file:Suppress("UnstableApiUsage")
-
-import com.android.build.gradle.AppPlugin
-import com.android.build.gradle.BaseExtension
-import com.android.build.gradle.LibraryExtension
-import com.android.build.gradle.LibraryPlugin
-import com.android.build.gradle.internal.tasks.factory.dependsOn
-import com.jfrog.bintray.gradle.BintrayExtension
-import com.jfrog.bintray.gradle.BintrayExtension.PackageConfig
-import com.jfrog.bintray.gradle.BintrayExtension.VersionConfig
 import io.gitlab.arturbosch.detekt.Detekt
-import org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper
-import org.jetbrains.kotlin.gradle.plugin.KotlinPluginWrapper
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
     /**
      * https://docs.gradle.org/current/userguide/base_plugin.html
      * base plugin added to add wiring on check->build tasks for detekt
      */
     base
+    id("io.gitlab.arturbosch.detekt")
+    id("com.autonomousapps.dependency-analysis") version "0.55.0" apply false
+    id("com.avito.android.libraries")
+
+    // workaround to load plugin classes once:
+    // https://youtrack.jetbrains.com/issue/KT-31643#focus=Comments-27-3510019.0-0
     id("org.jetbrains.kotlin.jvm") apply false
     id("com.android.application") apply false
-    id("io.gitlab.arturbosch.detekt")
-    id("com.autonomousapps.dependency-analysis") apply false
-    id("com.jfrog.bintray") version "1.8.4" apply false
+    id("com.jfrog.bintray") apply false
 }
 
 buildscript {
     configurations.classpath {
         resolutionStrategy {
             // com.autonomousapps.dependency-analysis depends on older version of okio, and it's resolved for
             // our instrumentation-tests plugin in subprojects in runtime
-            force(Dependencies.okio)
+            force("com.squareup.okio:okio:2.7.0")
```

Comment:
mmm
`Gradle` : "we make plugins for reusing versions"

Reply:
i'll fix it later